controllers/userStatus.js

```diff
@@ -36,20 +36,28 @@ const deleteUserStatus = async (req, res) => {
  */
 const getUserStatus = async (req, res) => {
   try {
-    const { userId } = req.params;
-    const userData = await userStatusModel.getUserStatus(userId);
-    const { userStatusExists, id, data } = userData;
-    let responseObject = { id, userId };
-    if (data) responseObject = { ...responseObject, ...data };
-    let statusCode;
-    if (userStatusExists) {
-      statusCode = 200;
-      responseObject.message = "User Status found successfully.";
+    let userId;
+    if (req.route.path === "/self") {
+      userId = req.userData.id;
     } else {
-      responseObject.message = "User Status couldn't be found.";
-      statusCode = 404;
+      userId = req.params.userId;
+    }
+    if (userId) {
+      const userData = await userStatusModel.getUserStatus(userId);
+      const { userStatusExists, id, data } = userData;
+      let responseObject = { id, userId };
+      if (data) responseObject = { ...responseObject, ...data };
+      let statusCode;
+      if (userStatusExists) {
+        statusCode = 200;
+        responseObject.message = "User Status found successfully.";
+      } else {
+        responseObject.message = "User Status couldn't be found.";
+        statusCode = 404;
+      }
```

Comment:
I agree with your point but since I am not adding `data` as a direct key here instead I am destructuring the `data` object so that all the keys of `data` become the primary keys of `responseObject` as well. So if the `data` object exist then only those keys will be added if not then we are adding a relevant message `User Status couldn't be found.`

Reply:
I was thinking if we can have something like this. 

{ data: ...data} (where data is the key and we destructure the data obj from response) if the data in response obj does not exist we pass null instead along with the custom message.

Reply:
since `data` itself is an object. executing `{ data: ...data}` will throw syntax error. the other choice we have is to pass the entire object as `{data: data}` but it will nest the status object 1 level inside the `data` key instead of directly being accessible at the parent level.

Reply:
yeah, it would be better to add 1 level of nesting to have a consistent schema. Updated in the latest code.